Assets/Scripts/Game/GameManager.cs

```diff
@@ -59,23 +61,43 @@ private IEnumerator HandleWallsDangerousness()
             {
                 yield return null;
 
-                while (!_waveManager.CurrentWave.switchOnWalls)
+                MoveRestingWallsToDangerous();
+
+                while (_waveManager.CurrentWave.dangerWallAmount == 0 || 
+                    _waveManager.CurrentWave.dangerWallAmount <= _dangerWallsCurrently ||
+                    _wallsToBeDangerous.Count == 0)
                 {
+                    MoveRestingWallsToDangerous();
                     yield return new WaitForSeconds(1);
                 }
 
-                var dangerousInterval = _wallDangerousInterval + _wallWarningInterval;
-
-                var wall = WallsToBeDangerous[new Random().Next(WallsToBeDangerous.Count)];
-                
-                StartCoroutine(wall.GetComponent<WallDanger>().BecameDangerousCoroutine(_wallWarningInterval));
+                var wall = _wallsToBeDangerous[new Random().Next(_wallsToBeDangerous.Count)];
+                _dangerWallsCurrently++;
+                wall.CanBeDangerous = false;
```

Comment:
This should be done inside `BecameDangerousCoroutine` otherwise the details of implementation are exposed.

Reply:
Agree